### DISCUSSION TITLE
How do I create table caption and numbering without a code chunk?

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#5223] [Q&A] [answered]
How do I create table caption and numbering without a code chunk? Say I have a table just manually pasted.

```
  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
1          5.1         3.5          1.4         0.2  setosa
2          4.9         3.0          1.4         0.2  setosa
3          4.7         3.2          1.3         0.2  setosa
4          4.6         3.1          1.5         0.2  setosa
5          5.0         3.6          1.4         0.2  setosa
6          5.4         3.9          1.7         0.4  setosa
```
: This is a caption {#tbl-iris}

Figures that are not generated from code can be captioned and numbered like this:

`![Surus](surus.png){#fig-surus}`

Is there something similar for table not generated from code?

**Comment by cscheid** [comment]
We have documentation on that: https://quarto.org/docs/authoring/tables.html

**Reply by royfrancis** [reply]
I have seen that, but, is it necessary that the table is in markdown format? If I put this `: This is a caption {#tbl-iris}` randomly somewhere it doesn't work.

**Reply by cderv** [reply]
> Is there something similar for table not generated from code?  (...) is it necessary that the table is in markdown format?

You want to include some tables not produce by code. Which is the format you want to use if not Markdown ? 

> If I put this : This is a caption {#tbl-iris} randomly somewhere it doesn't work.

This is a Markdown syntax construct, and it works for Markdown tables. You have the Quarto documentation, and also [Pandoc's Manual](https://pandoc.org/MANUAL.html#tables)https://pandoc.org/MANUAL.html#tables) about that

**Comment by ghost** [comment]
@royfrancis Can you please provide us with a working example so we can try and reproduce what you are seeing?  Thank you.

**Comment by royfrancis** [comment]
Here is an example:

````
---
title: Testing table
---

## Code

```{r}
#| label: tbl-code
#| tbl-cap: Table from code

head(iris)
```

## Markdown

|#|Sepal.Length|Sepal.Width|Petal.Length|Petal.Width|Species|
|:----|:----|:----|:----|:----|:----|
|1|5.1|3.5|1.4|0.2|setosa|
|2|4.9|3.0|1.4|0.2|setosa|
|3|4.7|3.2|1.3|0.2|setosa|
|4|4.6|3.1|1.5|0.2|setosa|
|5|5.0|3.6|1.4|0.2|setosa|
|6|5.4|3.9|1.7|0.4|setosa|

: Markdown table {#tbl-markdown}

## Text

  Sepal.Length Sepal.Width Petal.Length Petal.Width Species  
1          5.1         3.5          1.4         0.2  setosa  
2          4.9         3.0          1.4         0.2  setosa  
3          4.7         3.2          1.3         0.2  setosa  
4          4.6         3.1          1.5         0.2  setosa  
5          5.0         3.6          1.4         0.2  setosa  
6          5.4         3.9          1.7         0.4  setosa  

: Table regular text {#tbl-text}

## Text (pre)

```
  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
1          5.1         3.5          1.4         0.2  setosa
2          4.9         3.0          1.4         0.2  setosa
3          4.7         3.2          1.3         0.2  setosa
4          4.6         3.1          1.5         0.2  setosa
5          5.0         3.6          1.4         0.2  setosa
6          5.4         3.9          1.7         0.4  setosa
```

: Table code styled {#tbl-pre}

````
![Screenshot from 2023-04-18 10-44-29](https://user-images.githubusercontent.com/5083500/232723546-fffb840d-0d7f-403a-876e-8f9ac9c5a95d.png)

`quarto 1.2.335`

**Reply by cderv** [reply, marked as answer]
The last two are not table - they are not using a Table syntax, and so can't be parsed as table. First one is normal text, and second one if codeblock - nothing indicates that they are table. Quarto uses Pandoc's markdown and its feature as described in [Markdown Basics](https://quarto.org/docs/authoring/markdown-basics.html) and other examples. Same way as you need to use `#` to create a header, you need to follow the expected syntax for tables. The syntax is a way to indicate to Quarto and Pandoc that the text is indeed a Table and should be parsed and handled as such.  The `: This is a caption {#tbl-iris}` syntax is part of the Table Markdown parsing. 

Additionnaly to Markdown, HTML table will be parsed - LaTeX tables can be used for PDF document only. 

But arbitrary table as delimited text needs to be processed to be transformed to tables. 
In our doc, we provide a few link to tools that will help authoring table from text. https://quarto.org/docs/authoring/markdown-basics.html
In R ecosytem, a tool like [**datapasta**](https://milesmcbain.github.io/datapasta/) in R ecosytem) can be used to import some table data into R object, which you could modify and print using code chunk. Otherwise, tools like [**readr**](https://readr.tidyverse.org/) will help you import delimited text as tables. 

Hope it helps

**Reply by royfrancis** [reply]
Ok, I see. So you are saying it has to be a table generated from code or a markdown table. I cannot create an arbitrary caption.
